engine/orchestration/src/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1353,6 +1363,20 @@ public void prepare(VirtualMachineProfile vm, DeployDestination dest) throws Sto
             disk.setDetails(getDetails(volumeInfo, dataStore));
 
             vm.addDisk(disk);
+
+            // If hypervisor is vSphere, check for clone type setting.
+            if (vm.getHypervisorType().equals(HypervisorType.VMware)) {
+                // retrieve clone flag.
+                UserVmCloneType cloneType = UserVmCloneType.linked;
+                Boolean value = CapacityManager.VmwareCreateCloneFull.valueIn(vol.getPoolId());
+                if (value != null) {
+                    if (value.booleanValue() == true)
```

Comment:
Please collapse lines lines 1372 and 1373 into ``if (value != null && value)``.  The compiler will autobox the ``Boolean`` instance -- removing the requirement to call ``booleanValue``.